extract_CRISPRdb.py

```diff
@@ -0,0 +1,124 @@
+import requests
+from pattern import web
+import re
+import csv
+import time
+import os
+
+def get_dom(url):
+    html = requests.get(url).text
+    dom = web.Element(html)
+    return dom
+
+def get_taxons_from_CRISPRdb():
+    url = "http://crispr.u-psud.fr/crispr/"
+    dom_homepage = get_dom(url)
+    container = dom_homepage('div[class="strainlist"]')[0]
+    crisprs = []
+    for link in container('a'):
+        crispr = {}
+        crispr['Name'] = link.content.encode('ascii','ignore')
+        crispr['Taxon_id'] = link.href.encode('ascii','ignore')[46:]
+        crisprs.append(crispr)
+    return crisprs
+
+def get_genome_properties(crispr):
+    url = "http://crispr.u-psud.fr/cgi-bin/crispr/SpecieProperties.cgi?Taxon_id=" + crispr['Taxon_id']
+    dom = get_dom(url)
+    table = dom('table[class="primary_table"]')[1]
+    crispr['Ref_seqs']={}
+    for sequence in table('tr'):
+        if sequence('th')==[]:
+            crispr['Ref_seqs'][sequence('td')[1].content.encode('ascii','ignore')]=[]
+    return crispr
+
+def get_CRISPR_properties(crispr):
+    for ref_seq in crispr['Ref_seqs'].keys():
+        url = "http://crispr.u-psud.fr/crispr/CRISPRProperties.php?RefSeq=" + ref_seq + "&Taxon=" + crispr['Taxon_id']
+        dom = get_dom(url)
+        div = dom('div[class="rightcontent"]')[1]
+        table = div('table[class="primary_table"]')[0]
+        tbody = table('tbody')[0]
+        for crispr_id in tbody('tr'):
+            cell = crispr_id('td')[1]
+            crispr_id_value = cell('font')[0].content.encode('ascii','ignore').replace('<br/n/>','')
+            crispr_id_value = crispr_id_value.replace('\n','')
+            crispr_id_value = crispr_id_value.replace('\t','')
+            crispr_id_value = crispr_id_value.replace('<br />','')
+            crispr['Ref_seqs'][ref_seq].append(crispr_id_value)
+    return crispr
+
+def get_begin(source):
+    begin = re.search('(?<=Crispr_begin_position: )\d+', source)
```

Comment:
Here and in the `get_end` function, it might be good to tolerate more than one space in the regular expression. I found at least one example of the expression `'(?<=Crispr_begin_position: )\d+'` with three spaces before the number instead of one, and when this happened the script stopped.